sparrow-py/pysrc/sparrow_py/sources/arrow.py

```diff
@@ -49,11 +50,23 @@ def fix_nullable(field: pa.Field) -> pa.Field:
 
         self._ffi_table.add_pyarrow(data)
 
-    def add_data(self, data: Union[pd.DataFrame, pa.RecordBatch]) -> None:
+    def add_data(self, data: Union[pd.DataFrame, pa.RecordBatch, pa.Table]) -> None:
         """Add data to the source."""
         if isinstance(data, pd.DataFrame):
             data = pa.RecordBatch.from_pandas(data, self._schema, preserve_index=False)
-        self._ffi_table.add_pyarrow(data)
+
+        if isinstance(data, pa.RecordBatch):
+            self._ffi_table.add_pyarrow(data)
+        elif isinstance(data, pa.Table):
+            for batch in data.to_batches():
+                self._ffi_table.add_pyarrow(batch)
+        else:
+            raise TypeError(f"Unexpected type {type(data)}")
+
+    def add_parquet(self, path: str) -> None:
```

Comment:
This is a minimal "add a file by reading it into Parquet". We should discuss the organization of the source module to make things more convenient, but it works for now.